File: Jack.Extensions/DescricaoAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Extensions
{
    public class DescricaoAttribute : Attribute
    {
        private string _descricao;
        private string _valorReal;

        private int _id;

        public DescricaoAttribute()
        {
        }

        public DescricaoAttribute(string descricao)
        {
            this.Descricao = descricao;
        }

        public DescricaoAttribute(string descricao, string valorReal)
        {
            this.Descricao = descricao;
            this.ValorReal = valorReal;
        }
        public DescricaoAttribute(int id, string descricao, string valorReal)
        {
            this.Id = id;
            this.Descricao = descricao;
            this.ValorReal = valorReal;
        }

        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }
        public string Descricao
        {
            get { return _descricao; }
            set { _descricao = value; }
        }

        public string ValorReal
        {
            get { return _valorReal; }
            set { _valorReal = value; }
        }

        public override string ToString()
        {
            return _descricao;
        }
    }
}

File: Jack.Extensions2/DescricaoAttribute.vb

Public Class DescricaoAttribute
    Inherits Attribute
    Private _descricao As String
    Private _valorReal As String
    Private _id As Integer


    Public Sub New()
    End Sub

    Public Sub New(descricao As String)
        Me.Descricao = descricao
    End Sub

    Public Sub New(descricao As String, valorReal As String)
        Me.Descricao = descricao
        Me.ValorReal = valorReal
    End Sub
    Public Sub New(id As Integer, descricao As String, valorReal As String)
        Me.Id = id
        Me.Descricao = descricao
        Me.ValorReal = valorReal
    End Sub

    Public Property Id() As Integer
        Get
            Return _id
        End Get
        Set
            _id = Value
        End Set
    End Property
    Public Property Descricao() As String
        Get
            Return _descricao
        End Get
        Set
            _descricao = Value
        End Set
    End Property

    Public Property ValorReal() As String
        Get
            Return _valorReal
        End Get
        Set
            _valorReal = Value
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return _descricao
    End Function
End Class
